builtin/logical/transit/backend.go

```diff
@@ -47,23 +52,55 @@ func Backend(conf *logical.BackendConfig) *backend {
 			b.pathBackup(),
 			b.pathRestore(),
 			b.pathTrim(),
+			b.pathCacheConfig(),
 		},
 
 		Secrets:     []*framework.Secret{},
 		Invalidate:  b.invalidate,
 		BackendType: logical.TypeLogical,
 	}
 
-	b.lm = keysutil.NewLockManager(conf.System.CachingDisabled())
+	// determine cacheSize to use. Defaults to 0 which means unlimited
+	cacheSize := 0
+	useCache := !conf.System.CachingDisabled()
+	if useCache {
+		var err error
+		cacheSize, err = GetCacheSizeFromStorage(ctx, conf.StorageView)
+		if err != nil {
+			return nil, errwrap.Wrapf("Error retrieving cache size from storage: {{err}}", err)
+		}
+	}
 
-	return &b
+	var err error
+	b.lm, err = keysutil.NewLockManager(useCache, cacheSize)
+	if err != nil {
+		return nil, err
+	}
+
+	return &b, nil
 }
 
 type backend struct {
 	*framework.Backend
 	lm *keysutil.LockManager
 }
 
+func GetCacheSizeFromStorage(ctx context.Context, s logical.Storage) (int, error) {
+	size := 0
+	entry, err := s.Get(ctx, "config/cache-size")
```

Comment:
Is there a reason to make this entry so special purpose? Could it just be some kind of generic `config`/`transitConfig thing so that if we add more knobs later we can reuse the same struct?